cmd/nerdctl/run_mount_windows_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestRunVolume(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	imageName := testutil.WindowsNano
+	tID := testutil.Identifier(t)
+	rwDir, err := os.MkdirTemp(t.TempDir(), "rw")
+	if err != nil {
+		t.Fatal(err)
+	}
+	roDir, err := os.MkdirTemp(t.TempDir(), "ro")
+	if err != nil {
+		t.Fatal(err)
+	}
+	rwVolName := tID + "-rw"
+	roVolName := tID + "-ro"
+	for _, v := range []string{rwVolName, roVolName} {
+		defer base.Cmd("volume", "rm", "-f", v).Run()
+		base.Cmd("volume", "create", v).AssertOK()
+	}
+
+	containerName := tID
+	defer base.Cmd("rm", "-f", containerName).Run()
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", fmt.Sprintf("%s:C:\\mnt1", rwDir),
+		"-v", fmt.Sprintf("%s:C:\\mnt2ro", roDir),
+		"-v", fmt.Sprintf("%s:C:\\mnt3", rwVolName),
+		"-v", fmt.Sprintf("%s:C:\\mnt4ro", roVolName),
+		imageName).AssertOK()
+	base.Cmd("container", "inspect", containerName).AssertOK()
```

Comment:
this asserts that the container created, do we also want to make sure that the volume is actually there (maybe with something in it)?  How is this tested on Linux?

Reply:
TestRunAnonymousVolume and  TestRunNamedVolume checks whether the volume is mounted or not. For linux, we havent added any test as the problem was with the windows part.

Reply:
I guess we some coverage that the volume is created. Another thing noticed now, is it seem like we are also testing read/write behavior here. I would think we want to verify that the in the test.

Reply:
For verifying the read/write behavior, inspect should print the details regarding it. But in inspect for windows, it doesn't work.

```
D:\GolandProjects\github.com\nerdctlmst\cmd\nerdctl>nerdctl inspect 1f44fa83de
[
    {
        "Id": "1f44fa83de31867ec49d2639a15f4d04fa6003765660afb9e50c51e887e89815",
        "Created": "2022-04-28T11:03:48.0466636Z",
        "Path": "c:\\windows\\system32\\cmd.exe",
        "Args": null,
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Pid": 25496,
            "ExitCode": 0,
            "FinishedAt": "2022-04-28T11:03:50.2407926Z"
        },
        "Image": "mcr.microsoft.com/windows/nanoserver:20H2",
        "ResolvConfPath": "",
        "HostnamePath": "",
        "LogPath": "",
        "Name": "nanoserver-1f44f",
        "Driver": "windows",
        "Platform": "windows",
        "AppArmorProfile": "",
        "Mounts": [
            {
                "Source": "C:\\ProgramData\\nerdctl\\052055e3\\volumes\\default\\testVolume\\_data",
                "Destination": "C:\\src",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
        "NetworkSettings": {
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "MacAddress": "",
            "Networks": {}
        }
    }
]
```

Reply:
This is the output Im getting.

Reply:
So the inspect output for the volume is there but the rw/ro info isn't valid?  Does the actual directory in the container have the correct attributes?   If it is just inspect that isn't return properly lets create an issue to resolve later.

Reply:
This issue is only with windows where I cant get the output for read/write behavior.